cmd/uhc/cluster/list/list.go

```diff
@@ -85,25 +83,79 @@ func prettyPrint(arrStr ...[]string) {
 }
 
 // updateRowPad updates the length of all strings in a given list
-// to match wanted length
-func updateRowPad(columnPad []int, st []string) []string {
-	if len(columnPad) == len(st) {
-		for i := range st {
-			// Add padding
-			if len(st[i]) < columnPad[i] {
-				st[i] = st[i] + strings.Repeat(" ", columnPad[i]-len(st[i]))
-				// Clip
-			} else {
-				st[i] = st[i][:columnPad[i]]
-			}
+// to match wanted length.
+func updateRowPad(columnPad []int, columnList []string) []string {
+	// Make sure padding list is at least as long as row list
+	st := columnList
+	fixLen := len(columnPad) - len(st)
+	if fixLen < 0 {
+		// Get last value of column pad and simply re-use it to fill columnPad up
+		valueToUse := columnPad[len(columnPad)-1]
+		for i := 0; i < fixLen*(-1); i++ {
+			columnPad = append(columnPad, valueToUse)
+		}
+	}
+	for i := range st {
+		// Add padding
+		if len(st[i]) < columnPad[i] {
+			st[i] = st[i] + strings.Repeat(" ", columnPad[i]-len(st[i]))
+			// Clip
+		} else {
+			st[i] = st[i][:columnPad[i]-2] + "  "
 		}
-	} else {
-		fmt.Fprintf(os.Stderr, "Slices do not have the same length\n")
-		os.Exit(1)
 	}
 	return st
 }
 
+// findMapValue will find a key and retrieve its value from the given map. The key has to be
+// a string and can be multilayered, for example `foo.bar`. Returns the value and a boolean
+// indicating if the value was found.
+func findMapValue(data map[string]interface{}, key string) (string, bool) {
+
+	// Split key into array
+	keys := strings.Split(key, ".")
+
+	// loop though elements in sliced string:
+	for _, element := range keys {
+
+		// if key is found, continue:
+		if val, ok := data[element]; ok {
+
+			switch typed := val.(type) {
+
+			// If key points to string:
+			case string:
+				return typed, true
+
+			// If key points to interface insance:
+			case map[string]interface{}:
+				data = typed
+
+			// If key points to an integer:
+			case int:
+				return strconv.Itoa(typed), true
+
+			// If key points to a float:
+			case float32:
+				return fmt.Sprintf("%f", typed), true
+
+			case float64:
+				return fmt.Sprintf("%f", typed), true
+
+			// Not dealing with other possible datatypes:
+			default:
+				return key, false
```

Comment:
This means that if there is a type that you don't expect (like `bool` for example) you will be printing the name of the column instead of the value. Is that what you want to do? I'd say it would be better to print nothing, or some indication that there is no value.